src/sentry/api/endpoints/organization_traces.py

```diff
@@ -0,0 +1,160 @@
+            for row in span_results["data"]:
+                spans_by_trace[row["trace"]].append(row)
+
+            trace_spans_count = sorted(
+                [(trace, len(spans)) for trace, spans in spans_by_trace.items()],
+                key=lambda item: item[0],
+                reverse=True,
+            )[:per_page]
+
+            spans_by_trace = {trace: spans_by_trace[trace] for trace, _ in trace_spans_count}
+
+            min_timestamp = snuba_params.end
+            max_timestamp = snuba_params.start
+            assert min_timestamp is not None
+            assert max_timestamp is not None
+            for spans in spans_by_trace.values():
+                for span in spans:
+                    timestamp = datetime.fromisoformat(span["timestamp"])
+                    if timestamp < min_timestamp:
+                        min_timestamp = timestamp
+                    if timestamp > max_timestamp:
+                        max_timestamp = timestamp
+
+            # TODO: move to use `update_snuba_params_with_timestamp`
+            time_buffer = options.get("performance.traces.transaction_query_timebuffer_days")
+            buffer = timedelta(days=time_buffer)
+            params["start"] = min_timestamp - buffer
+            params["end"] = max_timestamp + buffer
+            snuba_params.start = min_timestamp - buffer
+            snuba_params.end = max_timestamp + buffer
+
+            with handle_query_errors():
+                builder = SpansIndexedQueryBuilder(
+                    Dataset.SpansIndexed,
+                    cast(ParamsType, params),
+                    snuba_params=snuba_params,
+                    query=f"trace:[{', '.join(spans_by_trace.keys())}]",
+                    selected_columns=["trace", "count()", "trace_name()", "elapsed()"],
+                    limit=len(spans_by_trace),
+                    offset=0,
+                    limitby=("trace", serialized["maxSpansPerTrace"]),
+                    config=QueryBuilderConfig(
+                        functions_acl=["trace_name", "elapsed"],
+                        transform_alias_to_input_format=True,
+                    ),
+                )
+                trace_results = builder.run_query(Referrer.API_TRACE_EXPLORER_TRACES_META.value)
+                trace_results = builder.process_results(trace_results)
+
+            traces: list[TraceResult] = [
+                {
+                    "trace": row["trace"],
+                    "numSpans": row["count()"],
+                    "name": row["trace_name()"],
+                    "duration": row["elapsed()"],
+                    "spans": [
+                        {field: span[field] for field in serialized["field"]}
+                        for span in spans_by_trace[row["trace"]]
+                    ],
+                }
+                for row in trace_results["data"]
+            ]
+
+            return traces
```

Comment:
Return a dict w/  content and `meta` to be consistent, I think. Otherwise looks good for now.

Reply:
I added a meta for the span columns. The trace columns I've left alone as they're clearly typed, and not dynamic.